pandas/core/indexes/datetimelike.py

```diff
@@ -651,6 +652,15 @@ def __add__(self, other):
                 raise TypeError("cannot add {typ1} and {typ2}"
                                 .format(typ1=type(self).__name__,
                                         typ2=type(other).__name__))
+            elif isinstance(other, np.ndarray):
+                if is_timedelta64_dtype(other):
+                    return self._add_delta(TimedeltaIndex(other))
```

Comment:
you can only accept timedelta64 here, integer is not valid either fro datetimelikes in add

Reply:
As noted in another comment above, the integer case is being used by internals for ``PeriodIndex``. Example: [`pandas/tseries/offsets.py:_beg_apply_index#L468`](https://github.com/pandas-dev/pandas/blob/master/pandas/tseries/offsets.py#L468) adds `base_period (PeriodIndex) + roll (ndarray[dtype=int64])`.

Reply:
no we don't allow this user facting; this is ONLY an internal operation

Reply:
Sure, but I don't quite understand what you're getting at -- are you saying there's still something to be done here? I just added the integer branch to prevent aforementioned internally used operation from breaking. (since it returns `NotImplemented` as it currently does, it will fall through to ufunc implementation.) I can add a code comment to explain the intent?